pymeasure/display/widgets/table_widget.py

```diff
@@ -0,0 +1,546 @@
+        self._data = self.results.data
+        current_row_count, columns = self._data.shape
+        if (self.last_row_count < current_row_count):
+            # Request cells content update
+            self.data_changed.emit(self.last_row_count, 0,
+                                   current_row_count - 1, columns - 1)
+            self.last_row_count = current_row_count
+
+    def set_color(self, color):
+        self.color = color
+
+
+class PandasModelBase(QtCore.QAbstractTableModel):
+    """ This class provided a model to manage multiple panda dataframes and
+    display them as a single table.
+
+    The multiple pandas dataframes are provided as ResultTable class instances
+    and all of them share the same number of columns.
+
+    There are some assumptions:
+    - Series in the dataframe are identical, we call this number k
+    - Series length can be different, we call this number l(x), where x=1..n
+
+    The data can be presented as follow:
+    - By column: each series in a separate column, in this case table shape
+    will be: (k*n) x (max(l(x) x=1..n)
+    - By row: column fixed to the number of series, in this case table shape
+    will be: k x (sum of l(x) x=1..n)
+    """
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.results_list = []
+        self.row_count = 0
+        self.column_count = 0
+
+    def add_results(self, result):
+        if result not in self.results_list:
+            self.results_list.append(result)
+            result.data_changed.connect(partial(self._data_changed, result))
+            self.layoutChanged.emit()
+            result.init()
+            result.start()
+            result.update_data()
+
+    def remove_results(self, result):
+        self.results_list.remove(result)
+        self.row_count = self.pandas_row_count()
+        self.column_count = self.pandas_column_count()
+        result.stop()
+        self.layoutChanged.emit()
+
+    def rowCount(self, parent=None):
+        return self.row_count
+
+    def columnCount(self, parent=None):
+        return self.column_count
+
+    def data(self, index, role=QtCore.Qt.ItemDataRole.DisplayRole):
+        if index.isValid() and role in (QtCore.Qt.ItemDataRole.DisplayRole, SORT_ROLE):
+            result, row, col = self.translate_to_local(index.row(), index.column())
+            try:
+                value = result.data.iloc[row][col]
+                column_type = result.data.dtypes[col]
+                # Cast to column type
+                value_render = column_type.type(value)
+            except IndexError:
+                value = NaN
+                value_render = ""
+            if isinstance(value_render, float64):
+                # limit maximum number of decimal digits displayed
+                value_render = f"{value_render:.{result.float_digits:d}f}"
+
+            if role == QtCore.Qt.ItemDataRole.DisplayRole:
+                return (str(value_render))
```

Comment:
Minor thing: parentheses are redundant

Reply:
Done in commit https://github.com/pymeasure/pymeasure/pull/771/commits/3f10ceffcb01c30d739c308fbaa1b30a498135b8